src/server/services/readMamacData.js

```diff
@@ -26,12 +30,19 @@ async function readMamacData(meter) {
 	if (!meter.id) throw new Error(`${meter} doesn't have an id to associate readings with`);
 	const rawReadings = await reqPromise(`http://${meter.ipAddress}/int2.csv`);
 	const parsedReadings = await parseCsv(rawReadings);
-	return parsedReadings.map(raw => new Reading(
+	return parsedReadings.map(raw => {
+		console.log(raw);
+		const reading = parseInt(raw[0]);
+		if (isNaN(reading)) {
+			throw new Error(`Meter reading parses to NaN for meter named ${meter.name} with id ${meter.id}`);
```

Comment:
You should include the raw string that was supposed to be an int in this message.